### PR TITLE
Rewrite/pr4 frontend core

### PR DESCRIPTION
**Pull request by tomvoelker**
## Intent of Changes
- Rewrite frontend core for BibSonomy (Vue 3) to a modern, strictly-typed architecture.
- Provide project scaffolding and developer tooling: Vite, TypeScript (strict), ESLint, Prettier, Tailwind/PostCSS, MSW test/dev config, and documentation (CLAUDE.md + READMEs).
- Establish core app infrastructure: main.ts app bootstrap, router with auth guards, Pinia stores (auth, preferences, UI), i18n (de/en), and Vue Query integration.
- Deliver a reusable UI component library and pages: layout (Header, Navbar, Footer, Sidebar, PageContainer), UI primitives (Button, Input, Badge, Card, Spinner, IconButton, Link, etc.), post components (PostCard, BookmarkCard, PublicationCard, PostActions, PostMeta, PostTags, PostThumbnail), and pages (Home, Error404/500, NotImplemented).
- Implement API layer and data fetching: Axios client with interceptors, Zod schemas for runtime validation, composables (usePosts, useBranding, useTheme), and Vue Query-based CRUD composables.
- Add MSW-based mocks and test helpers (handlers, mock data, server/browser setup) for development and Vitest test environment.

## Main Risks
- Authentication & security
  - Basic auth header assembled from localStorage (auth_user/auth_token); may not match backend expectations and could leak credentials if storage is compromised.
  - 401 handling clears auth without recovery flow or user-facing session-expired UX.
  - Risk of accidentally enabling MSW in production if VITE_ENABLE_MOCKS misconfigured.
- Backward compatibility
  - Large rewrite; existing frontend integrations or customizations will be incompatible.
  - Strong Zod schemas + strict TS may surface runtime failures if backend responses differ from schemas.
  - Router guard behavior and route meta changes may redirect traffic unexpectedly.
- Data integrity & mocks
  - Hardcoded mock data and handlers may diverge from real API semantics (shapes, pagination, error cases), masking integration bugs.
  - PUT/POST behavior in mocks and types (CreatePostRequest vs UpdatePostRequest) may not match server partial-update behavior.
- State & persistence
  - Pinia stores persist to localStorage; conflicts may arise with server-side sessions or multi-device expectations.
  - preferences store attaches media query listeners and exposes $dispose(); improper lifecycle handling could leak listeners.
- Internationalization & UX
  - Default locale is German; deployments expecting English-first UX may be affected.
  - Missing translation keys or mismatched keys across locales could cause visible fallbacks.
- Testing & CI
  - New strict TS and lint rules may break CI until codebase and configs are fully aligned.
  - MSW/unhandled-request policies in tests require maintaining handler coverage.

## Follow-Up Work
- API contract verification: reconcile Zod schemas and composable expectations with backend API (CreatePost/UpdatePost shapes, pagination semantics).
- Secure auth flow: confirm auth header format, add user-facing session/reauth UX, and avoid storing sensitive tokens insecurely.
- MSW & mock parity: expand mock coverage to mirror backend edge cases and keep mock data in sync with API changes; ensure VITE_ENABLE_MOCKS defaults false in production.
- Accessibility audit: run a11y tests (contrast, keyboard nav, aria attributes) across components (segmented controls, icon buttons, alerts, inputs).
- Error UX: add user-facing error handling for network/500/validation failures beyond 401 handling (toasts or alerts).
- Tests: add unit tests for composables (usePosts/useBranding/useTheme), component tests for key UI pieces, and integration tests with MSW; ensure Vitest setup remains stable.
- Docs & onboarding: finalize developer docs, update README paths, and provide migration notes for consumers of previous frontend.
- Performance & build checks: audit bundle size, confirm lazy-loading and code-splitting, validate production build settings (no mocks, correct theme/locale defaults), and verify TypeScript strict compilation in CI.
- Cleanup & lifecycle: ensure Pinia stores and global listeners (media queries, toast timers) are properly disposed to prevent memory leaks.

### CODE REVIEW

**Comment by tomvoelker** [comment]
@coderabbitai review

File: bibsonomy-rest-api-v2/src/main/kotlin/org/bibsonomy/api/service/TagService.kt [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+    private fun decodeBasic(header: String): Pair<String, String> {
+        val base64Token = header.removePrefix("Basic ").trim()
+        val decoded = String(Base64.getDecoder().decode(base64Token), StandardCharsets.UTF_8)
+        val delim = decoded.indexOf(':')
+        require(delim >= 0) { "Invalid basic authentication token" }
+        val username = decoded.substring(0, delim)
+        val apiKey = decoded.substring(delim + 1)
+        return username to apiKey
+    }
```
[the remark this answers is not in the document]
Comment (tomvoelker) [reply bibsonomy-rest-api-v2/src/main/kotlin/org/bibsonomy/api/service/TagService.kt]: Fixed by merge from master - TagService now uses BasicAuthUtils.decode() consistently with PostService.

File: bibsonomy-rest-api-v2/src/main/kotlin/org/bibsonomy/api/config/ApiCorsConfig.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+@Configuration
+class ApiCorsConfig {
+
+    @Bean
+    fun corsConfigurationSource(): CorsConfigurationSource {
+        val config = CorsConfiguration().apply {
+            allowedOrigins = listOf(
+                "http://localhost:5173",
+                "http://localhost:4173"
+            )
```
[the remark this answers is not in the document]
Comment (tomvoelker) [reply bibsonomy-rest-api-v2/src/main/kotlin/org/bibsonomy/api/config/ApiCorsConfig.kt]: Fixed by merge from master - CORS origins are now configurable via `cors.allowed-origins` property with localhost defaults for development.

**Comment by tomvoelker** [comment]
@greptile

**Comment by tomvoelker** [comment]
@coderabbitai review

**Comment by tomvoelker** [comment]
@coderabbitai review
